### PR TITLE
libc:xtensa: move syscall to swint call

### PR DESCRIPTION
**Pull request by zhuyanlinzyl**
Signed-off-by: zhuyanlin <zhuyanlin1@xiaomi.com>

### CODE REVIEW

File: libs/libc/machine/xtensa/arch_setjmp.S [hunk 1]
```diff
@@ -127,8 +129,10 @@ setjmp:
 
   /* Flush registers.  */
   mov a4, a2      # save a2 (jmp_buf)
-  movi  a2, 0
-  syscall
+  movi  a2, SYS_flush_context
```
Comment (xiaoxiang781216) [inline libs/libc/machine/xtensa/arch_setjmp.S:132]: let's reuse full_contextsave and contextrestore?

Comment (Ouss4) [reply libs/libc/machine/xtensa/arch_setjmp.S:132]: Are any of these commands handled by the software interrupt?

Speaking of `setjmp`, please note that the current `longjmp` implementation can cause issues for code using `WINBASE` and `WINSTART` as they are not saved.  for ESP-IDF there is a patch here: https://github.com/espressif/esp-idf/blob/4350e6fef859b94f3efae6ceced751911c9d2cbc/components/esp_rom/patches/esp_rom_longjmp.S#L43-L59

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@Ouss4 do you have more concern about this patch?

**Comment by Ouss4** [comment]
@xiaoxiang781216 what does the `SYS_flush_context` command do?  We don't have it implemented.

**Comment by zhuyanlinzyl** [comment]
hi, @Ouss4 : I'm sorry `SYS_flush_context` is  only in our inner code. I have update in this patch now.

`SYS_flush_context` command will flush the window registers to stack. 
As window flush was done in interrupt handler, `SYS_flush_context` cmd action was empty in xtensa_swint

File: arch/xtensa/include/syscall.h [hunk 2]
```diff
@@ -119,14 +119,22 @@
 
 #define SYS_switch_context        (2)
 
+
```
Comment (xiaoxiang781216) [inline arch/xtensa/include/syscall.h:122]: remove the extra blank line

File: arch/xtensa/include/syscall.h [hunk 3]
```diff
@@ -145,22 +153,22 @@
...
-#define SYS_signal_handler        (6)
+#define SYS_signal_handler        (7)
 
 /* SYS call 7:
  *
  * void signal_handler_return(void);
  */
 
-#define SYS_signal_handler_return (7)
+#define SYS_signal_handler_return (8)
```
Comment (xiaoxiang781216) [inline arch/xtensa/include/syscall.h:171]: need update CONFIG_SYS_RESERVED

**Review by xiaoxiang781216 — APPROVED** [review]
LGTM, @Ouss4 please take a look.

**Review by Ouss4 — APPROVED** [review]
LGTM!

**Pull request merged**